packages/ember-runtime/lib/ext/rsvp.js

```diff
@@ -54,32 +31,39 @@ export function onerrorDefault(reason) {
     return;
   }
 
+  // seems like we should turn this into an error too?
   if (error && error.name !== 'TransitionAborted') {
-    if (Ember.testing) {
-      // ES6TODO: remove when possible
-      if (!Test && has(testModuleName)) {
-        Test = require(testModuleName)['default'];
-      }
-
-      if (Test && Test.adapter) {
-        Test.adapter.exception(error);
-        Logger.error(error.stack);
-      } else {
-        throw error;
-      }
-    } else if (Ember.onerror) {
-      Ember.onerror(error);
-    } else {
-      Logger.error(error.stack);
-    }
+    dispatchError(error);
```

Comment:
@stefanpenner - It is largely the same path, but instead of guarding all over for `Ember.Test` (for debug versions) we just register an `dispatchErrorOverride` method when `ember-testing` package loads.